### PR TITLE
Add caching to performance indicators.

### PR DESCRIPTION
**Pull request by jeffstieler**
This PR seeks to increase thespeed of the performance indicators endpoint by:
* Adding caching to the data retrieval
* Changing the request to use "end of the day" for the `before` param instead of "now" - giving caching a chance to work

### Detailed test instructions:

- Verify that performance indicators work as expected (change date ranges, etc)

### Changelog Note:

Performance: add caching to dashboard performance indicator requests.

### CODE REVIEW

File: client/dashboard/store-performance/index.js [hunk 1]
```diff
@@ -262,19 +262,13 @@ export default compose(
 
 		const primaryQuery = {
 			after: appendTimestamp( datesFromQuery.primary.after, 'start' ),
-			before: appendTimestamp(
```
Comment (becdetat) [inline client/dashboard/store-performance/index.js]: Why was this changed? How does this change behaviour?

Comment (jeffstieler) [reply client/dashboard/store-performance/index.js]: I changed this because it essentially made the request for performance indicators impossible to cache. Without the change, if your chosen date range ended on "today", the `before` timestamp became the value "now" - down to the second.

Since we're already [busting cache on sync](https://github.com/woocommerce/woocommerce-admin/search?q=invalidate&unscoped_q=invalidate) I decided to force this query to always be to the end of day (`T23:59:59`) so that the request would be the same, which is what the cache key is built off of.

**Comment by jeffstieler** [comment]
> Looks awesome. How long does the caching hold on to the values?

Without being invalidated by sync actions, the cache expiry is 1 week: https://github.com/woocommerce/woocommerce-admin/blob/4ca4e7063ada4f139fbd4a3b6a1b7282e52af571/src/API/Reports/Cache.php#L77

File: client/dashboard/store-performance/index.js (hunk 1 above)
Comment (becdetat) [reply client/dashboard/store-performance/index.js:265]: Ah of course, that makes sense. Thanks.
